cloudfoundry/provider/datasource_service_credential_binding_details.go

```diff
@@ -0,0 +1,192 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	cfv3client "github.com/cloudfoundry/go-cfclient/v3/client"
+	"github.com/cloudfoundry/terraform-provider-cloudfoundry/cloudfoundry/provider/managers"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ datasource.DataSource              = &ServiceCredentialBindingDetailsDataSource{}
+	_ datasource.DataSourceWithConfigure = &ServiceCredentialBindingDetailsDataSource{}
+)
+
+func NewServiceCredentialBindingDetailsDataSource() datasource.DataSource {
+	return &ServiceCredentialBindingDetailsDataSource{}
+}
+
+type ServiceCredentialBindingDetailsDataSource struct {
+	cfClient *cfv3client.Client
+}
+
+func (d *ServiceCredentialBindingDetailsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_credential_binding_details"
+}
+
+func (d *ServiceCredentialBindingDetailsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Gets information on Service Credential Binding Details of a particular type (`app` or `key`) for a given service instance.\n\n" +
+			"Prefer this over `datasource_service_credential_binding`, which will be deprecated going forward.\n\n" +
+			"If you want to fetch all credential bindings (both `app` and `key` types), use `datasource_service_credential_bindings` instead.",
```

Comment:
As mentioned above I would recommend:

"If you want to fetch all credential bindings (both `app` and `key` types), use datasource `cloudfoundry_service_credential_bindings` instead.","